cloudigrade/account/util.py

```diff
@@ -71,31 +74,48 @@ def save_instance_events(account, instance_data, region, events=None):
             instance_data, dict) else instance_data.instance_id,
         region=region,
     )
+
     if events is None:
         # Assume this is the initial event
+        machineimage = AwsMachineImage.objects.get(
+            ec2_ami_id=instance_data['ImageId']
+        )
         event = AwsInstanceEvent(
             instance=instance,
+            machineimage=machineimage,
             event_type=InstanceEvent.TYPE.power_on,
             occurred_at=timezone.now(),
             subnet=instance_data['SubnetId'],
-            ec2_ami_id=instance_data['ImageId'],
             instance_type=instance_data['InstanceType'],
         )
         event.save()
     else:
         for event in events:
+            machineimage = AwsMachineImage.objects.get(
+                ec2_ami_id=event['ec2_ami_id']
+            )
             event = AwsInstanceEvent(
                 instance=instance,
+                machineimage=machineimage,
                 event_type=event['event_type'],
                 occurred_at=event['occurred_at'],
                 subnet=event['subnet'],
-                ec2_ami_id=event['ec2_ami_id'],
                 instance_type=event['instance_type'],
             )
             event.save()
     return instance
 
 
+# def get_or_create_awsmachineimage(account, ec2_ami_id):
```

Comment:
why is this commented out?

Reply:
Good catch. This is leftover from a previous experiment. Will delete in the next commit!